src/PrefillGravityForms/Services/EncryptionService.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace OWC\PrefillGravityForms\Services;
+
+use Exception;
+
+class EncryptionService
+{
+    private static string $cipher = 'aes-256-gcm';
+    private const TAG_LENGTH = 16;
+
+    /**
+     * Encrypts the given data array into a secure string.
+     */
+    public static function encrypt(array $data): string
+    {
+        $iv = random_bytes(openssl_cipher_iv_length(self::$cipher));
+        $tag = '';
+        $plaintext = serialize($data);
+
+        $ciphertext = openssl_encrypt(
+            $plaintext,
+            self::$cipher,
+            static::getEncryptionKey(),
+            0,
+            $iv,
+            $tag,
+            '',
+            self::TAG_LENGTH
+        );
+
+        return base64_encode($iv . $tag . $ciphertext);
+    }
+
+    /**
+     * Decrypts the given string back into a data array.
+     */
+    public static function decrypt(string $encrypted): array
+    {
+        $raw = base64_decode($encrypted);
+
+        $ivLength = openssl_cipher_iv_length(self::$cipher);
+        $iv = substr($raw, 0, $ivLength);
+        $tag = substr($raw, $ivLength, self::TAG_LENGTH);
+        $ciphertext = substr($raw, $ivLength + self::TAG_LENGTH);
+
+        $plaintext = openssl_decrypt(
+            $ciphertext,
+            self::$cipher,
+            static::getEncryptionKey(),
+            0,
+            $iv,
+            $tag
+        );
+
+        return is_string($plaintext) ? unserialize($plaintext) : [];
```

Comment:
Als decrypten mislukt, dan is dat prima en geven we een lege array terug? Waarom is dat prima?